src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataOperationVisitor.cs

```diff
@@ -224,22 +226,51 @@ public override TaintedDataAbstractValue VisitObjectCreation(IObjectCreationOper
                     ProcessTaintedDataEnteringInvocationOrCreation(method, taintedArguments, originalOperation);
                 }
 
-                if (this.IsSanitizingMethod(method))
+                PooledHashSet<IsInvocationTaintedWithPointsToAnalysis> evaluateWithPointsToAnalysis = null;
+                PooledHashSet<IsInvocationTaintedWithValueContentAnalysis> evaluateWithValueContentAnalysis = null;
+                try
                 {
-                    return TaintedDataAbstractValue.NotTainted;
-                }
-                else if (this.DataFlowAnalysisContext.SourceInfos.IsSourceMethod(method))
-                {
-                    return TaintedDataAbstractValue.CreateTainted(method, originalOperation.Syntax, this.OwningSymbol);
+                    if (this.IsSanitizingMethod(method))
+                    {
+                        return TaintedDataAbstractValue.NotTainted;
+                    }
+                    else if (this.DataFlowAnalysisContext.SourceInfos.IsSourceMethod(
+                        method,
+                        out evaluateWithPointsToAnalysis,
+                        out evaluateWithValueContentAnalysis))
+                    {
+                        ImmutableArray<IArgumentOperation> argumentOperation = (originalOperation as IInvocationOperation).Arguments;
+                        IEnumerable<PointsToAbstractValue> pointsToAnalysisResult = argumentOperation.Select(o => GetPointsToAbstractValue(o.Value));
+                        IEnumerable<ValueContentAbstractValue> valueContentAnalysisResult = argumentOperation.Select(o => GetValueContentAbstractValue(o.Value));
+                        if ((evaluateWithPointsToAnalysis != null && evaluateWithPointsToAnalysis.Any(o => o(pointsToAnalysisResult)))
+                            || (evaluateWithValueContentAnalysis != null && evaluateWithValueContentAnalysis.Any(o => o(pointsToAnalysisResult, valueContentAnalysisResult))))
+                        {
+                            return TaintedDataAbstractValue.CreateTainted(method, originalOperation.Syntax, this.OwningSymbol);
+                        }
+
+                        return TaintedDataAbstractValue.NotTainted;
+                    }
+                    else if (visitedInstance != null && this.IsSanitizingInstanceMethod(method))
+                    {
+                        if (AnalysisEntityFactory.TryCreate(visitedInstance, out AnalysisEntity analysisEntity))
+                        {
+                            this.CurrentAnalysisData.SetAbstractValue(analysisEntity, TaintedDataAbstractValue.NotTainted);
+                        }
+
+                        return TaintedDataAbstractValue.NotTainted;
+                    }
                 }
```

Comment:
Sorry, one more thing, let's move this before the .IsSourceMethod() check, just to so we're always consistent with prioritizing sanitizers over sources.